### ISSUE TITLE
Is the .ffizer.hbs extension necessary?

### ISSUE THREAD

**Issue by kskyten** [ffizer/ffizer#78]
It seems to me like to only reason to treat file extensions differently is to avoid having to escape double braces in files without the extension.  Wouldn't it make more sense to expand templates in all files? I think this would also make existing cookiecutter templates mostly compatible with ffizer.

**Comment by davidB** [comment]
Yes, it's a volontary choice to avoid to run template engine over every files:
- better perfomence (template engine) is slower than raw copy
- avoid to parse binary files (or to have to manage some kind of configuration to list file to ignore)
- avoid to try to run template engine over template file that the project would like to keep as is (eg .hbs file of a web project template, or template for ffizer's templates (It seems I forgot to publish them, I'll do asap).

I though about ability to reuse cookiecutter templates as is, but could not be done also because cookiecutter include preset of variables (python oriented,...). I would like ffizer to be explicit (few preset) and as fast as possible.

Thanks for your question/suggestion.

**Comment by kskyten** [comment]
Good points.

- I have no idea of the performance characteristic, but my gut feeling is that the templates would have to be quite massive for there to be a big difference.
- This is a good point. Perhaps it is possible to detect this. One approach would be to blacklist files by extension.
- I guess it all boils down to whether you expect there to be more dynamic content vs. static or escapable content in your templates.

Would it be possible to make this configurable?

**Comment by davidB** [comment]
everything could be configurable, but I would like to avoid fragmentation in the way to define template, and to handle them (more code to maintain,...).

An other advantage of the explicit .ffizer.hbs extension is that your IDE or some tools doesn't try to read as a vaild .rs, .py, .xxx but as handlebar template.

About being able to used some templates from other engine, I though (but nothing done) about including a flag like '--format ck' and doing the conversion of the tempate from cookiecutter into ffizer format juste after download of the template during "apply" subcommand or via an other subcommand "convert", as is the phase of conversion is isolated (simpler to code, to test, to maintain, ...). But I don't have the bandwith to work on it, (currently I have some other crates that require my attention)

**Comment by davidB** [comment]
The `.ffizer.hbs` extension is something I like (and unique vs others that choose the backlist) when I browse the template I know what is a template what is not.

The downside, is that I created an other extension `.ffizer.raw` for `.ffizer.hbs` that should not be process by the template engine or for raw file that also exist as file to ignore (I admit it's also a workaround because template are not in a subfolder).

**Comment by hverlin** [comment]
For `java`, the package name needs to be replaced in all files. A workaround is to use `scripts` to do a global replace at the end, but ideally, it would make it easier to convert an existing template if we could use an optional suffix?

[copier](https://copier.readthedocs.io/en/) has an option for this: https://copier.readthedocs.io/en/stable/configuring/#templates_suffix

**Comment by davidB** [comment]
It's a good idea to add an options. I could not change the default behavior because not managing every files as template:

- is a boost in rendering (less transformations)
- avoids to convert, to escape non-template file that use `{{` (eg github workflow, helm chart,...)
- avoid issue on binary files
